ecl/hql/hqlgram2.cpp

```diff
@@ -12398,20 +12399,25 @@ void parseAttribute(IHqlScope * scope, IFileContents * contents, HqlLookupContex
     OwnedHqlExpr parsed = scope->lookupSymbol(name, LSFsharedOK|LSFnoreport, ctx);
     bool canCache = parsed && (parsed->getOperator() != no_forwardscope) && !alreadySimplified;
     bool isMacro = parsed && parsed->isMacro();
-    OwnedHqlExpr simplified;
     if (canCache && (ctx.syntaxChecking() || ctx.hasCacheLocation()))
     {
-        simplified.setown(createSimplifiedDefinition(parsed));
+        OwnedHqlExpr simplified = createSimplifiedDefinition(parsed);
 
         if (simplified)
         {
-            if (ctx.hasCacheLocation())
-                attrCtx.createCache(simplified, isMacro);
+            ctx.incrementAttribsSimplified();
+
+            if (ctx.hasCacheLocation() && attrCtx.createCache(simplified, isMacro))
+                ctx.incrementCacheAttribsCreated();
+
             if (ctx.checkSimpleDef())
                 verifySimpifiedDefinition(parsed, simplified, attrCtx);
+
             if (simplified != parsed && ctx.syntaxChecking() && !ctx.ignoreCache())
                 scope->defineSymbol(LINK(simplified));
         }
+        else
+            ctx.incrementAttribsNotSimplified();
```

Comment:
This is still not as useful as a total number of attributes processed
- it isn't set if syntax checking isn't happening
- I have to add 3 numbers (cache hits + simplified + notsimplified) instead of looking at 1 number.